UABC-227-Semester-bookingOpenDay-and-bookingOpenTime

tulza opened the pull request:
Added booking open day and time for CreateSemesterFormDialog and EditSemesterFormDialog

tulza committed:
Add Booking day and time to Edit semesters

- Modified semesters detail card to show booking day and time
- Added booking day and time to form schema
- form reset to current day and time
- added inputField for day and time

tulza committed:
Add booking day and time to create semesters

- Added booking day and time to hook form schema
- Added InputField for day and time

tulza committed:
minor style changes and gammar fix

dyzhuu commented on src/components/admin/semesters/CreateSemesterFormDialog.tsx:
```diff
@@ -172,6 +184,26 @@ export const CreateSemesterFormDialog = () => {
+            isError={!!errors.bookingOpenTime?.message}
+            errorMessage={errors.bookingOpenTime?.message}
+            autoComplete="off"
+            placeholder="09:00:00"
```
Can you make this of type time instead of text?

dyzhuu replied:
with this approach, you might have to do some extra steps in the mutation
```
startTime: `${data.startTime}:00`,
endTime: `${data.endTime}:00`,
```

dyzhuu commented on src/components/admin/semesters/CreateSemesterFormDialog.tsx:
```diff
@@ -172,6 +184,26 @@ export const CreateSemesterFormDialog = () => {
         </div>
+        <div className="flex gap-2 *:grow">
+          <TextInput
+            label="Open day"
```
change labels to say `Booking Open Day` and `Booking Open Time`

dyzhuu commented on src/components/admin/semesters/SemesterDetailCard.tsx:
```diff
@@ -30,6 +38,9 @@ const UnmemoizedSemesterDetailCard = () => {
         </OptionsPopover>
       </div>
+      <p className="underline decoration-secondary/80 decoration-1 underline-offset-4">
+        Open {bookingOpenDay.toLowerCase()} at {bookingOpenTime}
```
```suggestion
                Bookings open {bookingOpenDay} {convertTo12HourFormat(bookingOpenTime)}

```

dyzhuu replied:
Nicer formatting

tulza committed:
Improve UX + resolve conversation

- Changed input to type time
- Changed text formatting
- Changed label for booking
- Added splaceTime to utils to handle default values

tulza committed:
Merge branch 'main' into UABC-227-Semester-bookingOpenDay-and-bookingOpenTime

tulza committed:
removed test

dyzhuu commented on src/components/admin/semesters/CreateSemesterFormDialog.tsx:
```diff
@@ -36,6 +37,14 @@ const formSchema = z
       .min(1, "Field is required")
       .refine(validateDate, "Invalid date"),
+    bookingOpenDay: z
+      .string()
```
does z.enum() instead work here?

dyzhuu commented on src/components/admin/semesters/EditSemesterFormDialog.tsx:
```diff
@@ -76,6 +90,8 @@ export const EditSemesterFormDialog = () => {
       breakStart,
       breakEnd,
+      bookingOpenDay,
+      bookingOpenTime: spliceTime(bookingOpenTime),
```
maybe easier to just do .slice(0, 5) ?

dyzhuu reviewed (changes requested):
Looks good, just make those changes

dyzhuu commented on src/components/admin/semesters/CreateSemesterFormDialog.tsx:
```diff
@@ -172,6 +181,26 @@ export const CreateSemesterFormDialog = () => {
             placeholder="dd/MM/yyyy"
           />
         </div>
+        <div className="flex gap-2 *:grow">
```
Maybe move this div to the top, matches better with the card that way

tulza committed:
Style changes + minor refactor and fix

tulza committed:
remove splice import
